src/main/java/com/linecorp/armeria/server/ServiceMapping.java

```diff
@@ -111,9 +111,7 @@ public MappedService apply(String path) {
 
         // Cache miss or disabled cache
         MappedService result = MappedService.empty();
-        final int size = patterns.size();
-        for (int i = 0; i < size; i ++) {
-            final Entry<PathMapping, Service> e = patterns.get(i);
+        for (final Entry<PathMapping, Service> e : patterns) {
```

Comment:
The use of an index variable was intentional here to avoid unnecessary instantiation of an `Iterator`. Could you revert this?